utils.ts

```diff
@@ -38,6 +46,76 @@ export const makeQueryPeliasCompatible = (queryString: string): string => {
   return queryString.replace(/@/g, ' ').replace(/&/g, ' ')
 }
 
+export const hereResultTypeToPeliasLayer = (resultType: string): string => {
+  switch (resultType) {
+    case 'place':
+      return 'venue'
+    case 'houseNumber':
+      return 'address'
+    default:
+      return resultType
+  }
+}
+
+/**
+ * Executes a geocoder request using HERE API via @otp-ui/geocoder
+ * @param service Enum speicfying the type of API request to make.
+ * @param queryStringParam Query string from AWS with the url parameters from client.
+ * @param apiKey HERE API Key
+ * @returns HERE response in GeoJSON format.
+ */
+export const fetchHere = async (
+  service: string,
+  queryStringParams: Record<string, string>,
+  apiKey?: string
+): Promise<FeatureCollection> => {
+  const hereGeocoder = getGeocoder({
+    apiKey,
+    type: 'HERE'
+  })
+
+  const params = new URLSearchParams(queryStringParams)
+  const hereParams: AutocompleteQuery & SearchQuery & ReverseQuery = {}
+
+  // convert QSP into API call for hereGeocoder
+
+  const [minLat, minLon, maxLat, maxLon, size] = [
```

Comment:
Should most of these fixes not go into the geocoder package directly?

Reply:
I don't really think the geocoder package should handle this conversion from URL parmaeters, since it's pretty specific to this handler.

Reply:
Ah that makes sense. In that case I think my idea here would be that since both the Pelias endpoint and the HERE endpoint use the same geocoder package, the fetchPelias method should also be refactored to use this method/removed entirely. Doing that should allow us to remove a lot of code, right?

Reply:
Yes, that is a good idea. I hadn't done that becuase I was just trying to finish this project up and didn't want to take on another subtask. But now that you've said it, I will do it.